### PR TITLE
Fix sending color codes to clients that don't support them.

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -1644,14 +1644,21 @@ void Server::SendChatMessage(u16 peer_id, const std::wstring &message)
 {
 	DSTACK(FUNCTION_NAME);
 
+	std::wstring proccesed_message;
```
Comment (sfan5) [inline src/server.cpp]: processed

File: src/server.cpp [hunk 2]
```diff
@@ -1644,14 +1644,21 @@ void Server::SendChatMessage(u16 peer_id, const std::wstring &message)
...
-	pkt << message;
+	pkt << proccesed_message;
 
 	if (peer_id != PEER_ID_INEXISTENT) {
 		Send(&pkt);
 	}
 	else {
-		m_clients.sendToAll(&pkt);
+		for (u16 id: m_clients.getClientIDs())
+			SendChatMessage(message);
```
Comment (sfan5) [inline src/server.cpp]: this looks like it definitely doesn't do the right thing

Comment (red-001) [inline src/server.cpp]: yeah you are right, I have no idea how that even compiled.

Comment (red-001) [inline src/server.cpp]: fixed

File: src/server.cpp [hunk 3]
```diff
@@ -1644,14 +1644,21 @@ void Server::SendChatMessage(u16 peer_id, const std::wstring &message)
...
 	NetworkPacket pkt(TOCLIENT_CHAT_MESSAGE, 0, peer_id);
-	pkt << message;
+	pkt << processed_message;
 
 	if (peer_id != PEER_ID_INEXISTENT) {
 		Send(&pkt);
 	}
 	else {
-		m_clients.sendToAll(&pkt);
+		for (u16 id: m_clients.getClientIDs())
```
Comment (nerzhul) [inline src/server.cpp]: please change function to be properly recursive if needed, here we create a packet and verify the peer_id in broadcast mode, please move packet ceration into unicast mode and the proto version check

File: src/server.cpp [hunk 4]
```diff
@@ -1643,15 +1643,20 @@ void Server::SendInventory(PlayerSAO* playerSAO)
...
-
-	NetworkPacket pkt(TOCLIENT_CHAT_MESSAGE, 0, peer_id);
-	pkt << message;
-
 	if (peer_id != PEER_ID_INEXISTENT) {
+		std::wstring processed_message;
+		if (m_clients.getProtocolVersion(peer_id) < 27)
+			processed_message = unescape_enriched(message);
+		else
+			processed_message = message;
```
Comment (nerzhul) [inline src/server.cpp]: i forget to tell you to remove this temporary variable and write directly to packet in the condition members

Comment (red-001) [inline src/server.cpp]: ty

**Pull request merged**